core/src/test/java/org/apache/calcite/test/MaterializationTest.java

```diff
@@ -2468,12 +2468,25 @@ private void checkSatisfiable(RexNode e, String s) {
     checkMaterialize(sql, sql);
   }
 
-  @Test public void testUnionToUnion() {
+  @Test public void testUnionToUnion0() {
     String sql0 = "select * from \"emps\" where \"empid\" < 300";
     String sql1 = "select * from \"emps\" where \"empid\" > 200";
     checkMaterialize(sql0 + " union all " + sql1, sql1 + " union all " + sql0);
   }
 
+  @Test public void testUnionToUnion1() {
+    String sql0 = "select * from \"emps\" where \"empid\" < 300";
+    String sql1 = "select * from \"emps\" where \"empid\" > 200";
+    checkMaterialize(sql0 + " union " + sql1, sql1 + " union " + sql0);
+  }
+
+  @Test public void testUnionToUnion2() {
+    String sql0 = "select * from \"emps\" where \"empid\" < 300";
+    String sql1 = "select * from \"emps\" where \"empid\" > 200";
+    checkNoMaterialize(sql0 + " union " + sql1, sql0 + " union all " + sql1,
+        HR_FKUK_MODEL);
+  }
+
```

Comment:
If the MV is `union all`, but query is `union`, we can still materialize, right?

Reply:
Thanks  a lot, haisheng ~
Yes, when MV is `union all` but query is `union`, we can also succeed the matching if compensate a `distinct agg` outside of MV for deduplication.
But my current thinking is to keep the `UnionToUnionUnifyRule` simpler and don't bring in the deduplication logic. 
How do you think ? Which one do you prefer ?

Reply:
Make sense.